cake-autorate.sh

```diff
@@ -30,23 +30,23 @@ cleanup_and_killall()
 	log_msg "INFO" ""
 	log_msg "INFO" "Killing all background processes and cleaning up temporary files."
 
-	if ! [[ -z ${maintain_pingers_pid} ]]; then
+	if [[ -n ${maintain_pingers_pid} ]]; then
 		log_msg "DEBUG" "Terminating maintain_pingers_pid: ${maintain_pingers_pid}."
 		kill_and_wait_by_pid_name maintain_pingers_pid 0
 	fi
 
-	if ! [[ -z ${monitor_achieved_rates_pid} ]]; then
+	if [[ -n ${monitor_achieved_rates_pid} ]]; then
 		log_msg "DEBUG" "Terminating monitor_achieved_rates_pid: ${monitor_achieved_rates_pid}."
 		kill_and_wait_by_pid_name monitor_achieved_rates_pid 0
 	fi
 
-	if ! [[ -z ${maintain_log_file_pid} ]]; then
+	if [[ -n ${maintain_log_file_pid} ]]; then
 		log_msg "DEBUG" "Terminating maintain_log_file_pid: ${maintain_log_file_pid}."
 		kill_and_wait_by_pid_name maintain_log_file_pid 0
 	fi
 
 	[[ -d ${run_path} ]] && rm -r ${run_path}
-	[[ -d /var/run/cake-autorate ]] && compgen -G /var/run/cake-autorate/* > /dev/null || rm -r /var/run/cake-autorate
```

Comment:
Ah. But I think this extra test is needed because cake-autorate supports multiple instances running at the same time. With this change stopping one instance would break the other instances?

Reply:
It's worth noting that the `[[ -d .... ]]` at the start doesn't do anything. You could see this in: `false && true || echo Hello.`

Reply:
What's the use of compgen in this case?

Reply:
```
$ false && false || echo Hello.
Hello.
$ true && false || echo Hello.
Hello.
$ false && true || echo Hello.
Hello.
```

Reply:
Ah yes I see. I should have square bracketed?. The use of compgen is to verify that no other instances exist, instances are set as subdirectories within /var/run/cake-autorate like /primary/ and /secondary/ as set by the name of the identifier in the config e.g. cake-autorate_config.primary.sh.

Reply:
Ah this makes sense, you could use rmdir instead. `rmdir /var/lib/cake-autorate 2>/dev/null`.

Reply:
`rmdir` will remove the folder only if it is empty.